### PR TITLE
Closes #363 assert_true checks not correct. After discussion the whole method for…

### LINKED ISSUE #363: mcmc-GeneralData-OneParExpNormalPrior - assert check
There seems to be an error in the mcmc method and the correspnding description: 

https://github.com/Roche/crmPack/blob/d7cb6def0d43ba7cffbb439b367feda027934991/R/mcmc.R#L237-L239

https://github.com/Roche/crmPack/blob/d7cb6def0d43ba7cffbb439b367feda027934991/R/mcmc.R#L259-L260

The description should read:
```R
#'   [`OneParExpNormalPrior`] model, it is required that the length of
#'   skeleton prior probabilities vector should be equal to the length of the number of doses
```
and the code
```R
  if (!from_prior) {
      assert_true(length(model@skel_probs) == max(data@nGrid))
```
I am wondering if this check should not be done when the model is specified with the other model parameter validation
checks.

### PR DESCRIPTION
**Pull request by 0liver0815**
closes #363 

… OneParExpNormalPrior can be removed as not needed.

Tested with simulation. Works without the method in mcmc.

### CODE REVIEW

**Review by danielinteractive — COMMENTED** [review]
Thanks @0liver0815 , can you add a test as well where this is simulated, i.e. standard `mcmc()` is run for this model? or is it already included in codebase?

**Comment by 0liver0815** [comment]
@danielinteractive Yes it runs, but after thinking about the check of the slots in the model, I now believe that this check here is necessary. The reason is that it is possible that the user specifies the grid in the model, but the data may not fit to check and mcmc crash. So blease don't merge, I will change the code back and fix the check.

**Comment by danielinteractive** [comment]
@0liver0815 right, makes sense now. basically dose_grid is defined in the data outside the model therefore no guarantee that they match together well via the skeleton...

**Comment by danielinteractive** [comment]
@0liver0815 what is the next step here?

**Comment by 0liver0815** [comment]
@danielinteractive From my point of view this branch can already be merged. I can also add a test, before merging or the test can be added later.

File: R/mcmc.R
Comment (danielinteractive) [inline R/mcmc.R:260]:
```suggestion
      assert_true(length(model@skel_probs) == data@nGrid)
```

**Review by danielinteractive — APPROVED** [review]
just one simplification inside, afterwards good to merge

**Pull request merged**
